sdk/src/azure/iot/az_iot_hub_client_properties.c

```diff
@@ -65,7 +65,6 @@ AZ_NODISCARD az_result az_iot_hub_client_properties_writer_begin_component(
     az_json_writer* ref_json_writer,
     az_span component_name)
 {
-  _az_PRECONDITION_NOT_NULL(client);
```

Comment:
What motivated weakening this precondition?

Reply:
The ADU APIs need to use it without passing a hub client (since the adu client doesn't have access to it)